packages/markdown/index.ts

```diff
@@ -237,6 +237,26 @@ export function create(options: CreateOptions): ServicePlugin {
 					}
 				},
 
+				async provideDefinition(document, position, token) {
+					if (prepare(document)) {
+						let locations = await ls.getDefinition(document, position, token);
+
+						if (!locations) {
+							return;
+						}
+
+						if (!Array.isArray(locations)) {
+							locations = [locations];
+						}
+
+						return locations.map<LocationLink>(location => ({
+							targetUri: location.uri,
+							targetRange: location.range,
+							targetSelectionRange: location.range,
+						}));
```

Comment:
LSP allows [`textDocument/definitions`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_definition) to return `Location | Location[] | LocationLink[] | null`, so maybe Volar should allow this too instead of doing the his conversion.

Reply:
The ServicePlugin API required the only most demanding type because Volar needs to merge the results of multiple services behind the scenes.